src/main/java/org/rumbledb/items/parsing/ItemParser.java

```diff
@@ -274,6 +275,58 @@ private static void addValue(
         }
     }
 
+    public static DataType getDataFrameDataTypeFromItemTypeName(String itemTypeName) {
+        switch (itemTypeName) {
+            case "boolean":
+                return DataTypes.BooleanType;
+            case "integer":
+                return DataTypes.IntegerType;
+            case "double":
+                return DataTypes.DoubleType;
+            case "decimal":
+                return DataTypes.LongType;
+            case "string":
+                return DataTypes.StringType;
+            case "null":
+                return DataTypes.NullType;
+            case "date":
+                return DataTypes.DateType;
+            case "datetime":
+                return DataTypes.TimestampType;
+            case "hexbinary":
+                return DataTypes.BinaryType;
+            case "[\"item\"]":
+                return vectorType;
```

Comment:
This is a temporary representation for the vector types. Once we implement the support for array type representation, extra set of quotes will go away. If you have any preferences on how to represent vectors, please let me know

Reply:
From a data independence perspective, vectors are a hidden physical detail -- this representation is thus only a debug feature known from us only and users should not be able to create vectors directly (only VectorAssembler would do this from double fields, as well as other transformers and estimators that preprocess double fields to vectors to pass it on to Spark ML).

As such, this name is fine for now, or we could also call it "vector" -- but we should not document it anywhere in the end user documentation.

Reply:
I would much rather use the vector name here as well. But the problem is, then the user would be directly exposed the vector keyword. The reason being, annotate() function applies verifies the schema on the input dataframe with this new implementation instead of directly passing it along. Hence if users writes the following query:  
```
annotate(libsvm-file(....), {"label": "integer", "features": "?????" )
```
 They have to specify the correct type for features which maps to vectors. This is why I initially opted for passing dataframe as is, instead of applying a typecheck.

Reply:
After giving it some thought I have an idea for a different representation here. Data read by libsvm is always read into a SparseVector which is converted to an object in Rumble Data Model. Hence, how about we expose the keyword "object" or "{ }" in the short run?

Reply:
Your idea goes in the right direction, but let me think about this more. In the meantime, feel free to go ahead and merge this branch with the understanding that users do not know about this special type, and are only permitted to use annotate on flat objects with known atomic types, which is our main use case (which the output of libsvm is not).

In other words, let us just consider "[\"item\"]" a user-defined type known only to us -- in the schema language, type names can be any string anyway.


About this query:

annotate(libsvm-file(....), {"label": "integer", "features": "?????" )

Is there any scenario when the end user strictly has to use this, and otherwise would be incapable of executing a specific ML query? The output of libsvm-file can be directly processed by estimators and transformers and annotate leaves it unchanged anyway, correct?

Reply:
Precisely. This addition is more aimed at completeness rather than addressing a need. Libsvm data can be directly used in ML. However if the user decides to call annotate on it directly, which only serves to validate the schema, only then we need to cover the vector case.

Reply:
Perfect, thank you. Then it's all good.